.github/workflows/ci.yml

```diff
@@ -0,0 +1,148 @@
+name: CI
+
+on: [push]
+
+env:
+  IMAGE: ghcr.io/avtikhon/tarantool/testing/debian-stretch:latest
+  TRAVIS_MAKE: make -f .travis.mk
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    #runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+
+    container:
+      image: ghcr.io/avtikhon/tarantool/testing/debian-stretch:latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: test
+        run: ${TRAVIS_MAKE} test_debian_no_deps
+
+  debug:
+    runs-on: ubuntu-latest
+    #runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+
+    container:
+      image: ghcr.io/avtikhon/tarantool/testing/debian-stretch:latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: test
+        run: ${TRAVIS_MAKE} test_coverage_debian_no_deps
+        env:
+          COVERALLS_TOKEN: ${{ secrets.COVERALLS_TOKEN }}
+
+  release_clang:
+    runs-on: ubuntu-latest
+    #runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+
+    container:
+      image: ghcr.io/avtikhon/tarantool/testing/debian-stretch:latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: test
+        env:
+          CC: clang
+          CXX: clang++
+        run: ${TRAVIS_MAKE} test_debian_no_deps
+
+  release_lto:
+    runs-on: ubuntu-latest
+    #runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+
+    container:
+      image: ghcr.io/avtikhon/tarantool/testing/debian-buster:latest
+
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: test
+        env:
+          CMAKE_EXTRA_PARAMS: -DENABLE_LTO=ON
+        run: ${TRAVIS_MAKE} test_debian_no_deps
+
+  release_lto_clang11:
+    runs-on: ubuntu-latest
+    #runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+
+    container:
+      image: ghcr.io/avtikhon/tarantool/testing/debian-buster:latest
+
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: test
+        env:
+          CC: clang-11
+          CXX: clang++-11
+          CMAKE_EXTRA_PARAMS: -DENABLE_LTO=ON
+        run: ${TRAVIS_MAKE} test_debian_no_deps
+
+  release_asan_clang11:
+    runs-on: ubuntu-latest
+    #runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+
+    container:
+      image: ghcr.io/avtikhon/tarantool/testing/debian-buster:latest
+
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: test
+        run: ${TRAVIS_MAKE} test_asan_debian_no_deps
+
+  osx_10_15:
+    runs-on: macos-10.15
+    #runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: test
+        run: ${TRAVIS_MAKE} test_osx_github_actions
+
+  osx_11_0:
+    runs-on: macos-11.0
```

Comment:
I'd *highly* recommend to *not use* our runners for GitHub PR, and use generic macosx-latest instead. Our runners are limited in capacity, GitHub one's are unlimited (in theory). 

We actually do not want to reduce capability of our MacOSX CI farm because of flow of sudden increase of external pull-requests.

Reply:
I mean - create several .yml-files which would react to the same set of events, but which we could restart separately if there is failure.

Reply:
It will increase the number of entries of a single testing also luacheck stage would be duplicated, it will increase the testing time.

Reply:
There is no much point to proceed luacheck before each flow (as it was done in GitLab pipeline) - it might be done separately, in the separate flow. This approach still produces the same result.

Reply:
We don't need to test standalone jobs, we really need CI/CD process with pipelines and jobs as it's parts.

Reply:
This discussion was started because of the thesis that jobs use internal hosts for testing, the thesis was wrong - all jobs run on external hosts. Please resolve this discussion due to I don't see any more objections to divide pipeline on standalone jobs, or please provide anything that I didn't answer.